workflow/controller/dag.go

```diff
@@ -407,6 +412,8 @@ func (woc *wfOperationCtx) executeDAGTask(ctx context.Context, dagCtx *dagContex
 
 		if node.Completed() {
 			// Run the node's onExit node, if any.
+			prefix := fmt.Sprintf("tasks.%s", task.Name)
+			woc.buildLocalScope(scope, prefix, node)
 			hasOnExitNode, onExitNode, err := woc.runOnExitNode(ctx, task.GetExitHook(woc.execWf.Spec.Arguments), node, dagCtx.boundaryID, dagCtx.tmplCtx, "tasks."+taskName, scope)
 			if hasOnExitNode && (onExitNode == nil || !onExitNode.Fulfilled() || err != nil) {
```

Comment:
just want to make sure - I noticed `woc.runOnExitNode` is also called on line 562. Is that part good to go as far as the scope?